cache/disk/disk.go

```diff
@@ -40,21 +41,36 @@ type diskCache struct {
 func New(dir string, maxSizeBytes int64) cache.Cache {
 	// Create the directory structure.
 	hexLetters := []byte("0123456789abcdef")
+	dirs := make([]string, len(hexLetters)*len(hexLetters)*3)
+	idx := 0
+
 	for _, c1 := range hexLetters {
 		for _, c2 := range hexLetters {
 			subDir := string(c1) + string(c2)
-			err := os.MkdirAll(filepath.Join(dir, cache.CAS.String(), subDir), os.FileMode(0744))
+
+			casSubDir := filepath.Join(dir, cache.CAS.String(), subDir)
+			err := os.MkdirAll(casSubDir, os.FileMode(0777))
```

Comment:
I am not the expert but I figured it's good practice to be as restrictive as possible. @bdittmer what's the reason for changing this to 777?

Reply:
The use case is so that a group of users can admin the cache without root access. The permissions on disk are not 0777 but (0777 - umask) for dirs and (0666 - umask) for files.

I think this permission change is OK, but it should land in a separate commit. Added an alternative permissions change in #128 (which will cause a merge conflict here- sorry).

Reply:
No problem, thanks!